### PR TITLE
[TR] PIM-4265: fix static analysis and behats

### PR DESCRIPTION
**Pull request by sumbobyboys**
| Q                    | A
| -------------------- | ---
| Bug fix?             | Y
| New feature?         | N
| BC breaks?           | N
| CI currently passes? | N
| Tests pass?          | Y
| Scenarios pass?      | Y
| Checkstyle issues?*  | N
| PMD issues?**        | N
| Changelog updated?   | N
| Fixed tickets        | PIM-4265
| DB schema updated?   | N
| Migration script?    |
| Doc PR               |

### CODE REVIEW

**Comment by solivier** [comment]
GTM

**Comment by nidup** [comment]
GTM

File: features/Context/NavigationContext.php [hunk 1]
```diff
@@ -697,6 +701,8 @@ protected function assertAddress($expected)
 
     /**
      * A method that logs the user in with the previously provided credentials if required by the page
+     *
+     * @return boolean true if login was required, false if not
```
Comment (grena) [inline features/Context/NavigationContext.php]: I think we use `bool` now in phpdoc

**Comment by grena** [comment]
GTM

File: features/Context/NavigationContext.php [hunk 2]
```diff
@@ -659,8 +659,12 @@ public function openPage($page, array $options = [])
         $page = $this->getCurrentPage()->open($options);
 
         // spin function to deal with invalid CSRF problems
-        $this->getMainContext()->spin(function () use ($page) {
-            $this->loginIfRequired();
+        $this->getMainContext()->spin(function () use ($page, $options) {
+            if ($this->loginIfRequired()) {
+                $page = $this->getCurrentPage()->open($options);
+                $this->wait();
+                echo "login was required\n";
```
Comment (jmleroux) [inline features/Context/NavigationContext.php]: `echo "login was required" . PHP_EOL;`

**Comment by jmleroux** [comment]
GTM

**Pull request merged**